spec/coinbase/address_spec.rb

```diff
@@ -44,39 +45,103 @@
   end
 
   describe '#list_balances' do
-    before do
-      allow(client).to receive(:eth_getBalance).with(address_id, 'latest').and_return('0xde0b6b3a7640000')
-    end
-
-    it 'returns a hash with an ETH balance' do
-      expect(address.list_balances).to eq(eth: BigDecimal('1'))
+    let(:response) do
+      Coinbase::Client::AddressBalanceList.new(
+        'data' => [
+          Coinbase::Client::Balance.new(
+            {
+              'amount' => '1000000000000000000',
+              'asset' => Coinbase::Client::Asset.new({
+                                                       'network_id': 'base-sepolia',
+                                                       'asset_id': 'eth',
+                                                       'decimals': 18
+                                                     })
+            }
+          ),
+          Coinbase::Client::Balance.new(
+            {
+              'amount' => '5000',
+              'asset' => Coinbase::Client::Asset.new({
+                                                       'network_id': 'base-sepolia',
+                                                       'asset_id': 'usdc',
+                                                       'decimals': 6
+                                                     })
+            }
+          )
+        ]
+      )
+    end
+
+    it 'returns a hash with balances' do
+      expect(addresses_api)
+        .to receive(:list_address_balances)
+        .with(wallet_id, address_id)
+        .and_return(response)
+      expect(address.list_balances).to eq(eth: BigDecimal('1'), usdc: BigDecimal('5000'))
     end
   end
 
   describe '#get_balance' do
-    before do
-      allow(client).to receive(:eth_getBalance).with(address_id, 'latest').and_return('0xde0b6b3a7640000')
+    let(:response) do
+      Coinbase::Client::Balance.new(
+        {
+          'amount' => '1000000000000000000',
+          'asset' => Coinbase::Client::Asset.new({
+                                                   'network_id': 'base-sepolia',
+                                                   'asset_id': 'eth',
+                                                   'decimals': 18
+                                                 })
+        }
+      )
     end
 
     it 'returns the correct ETH balance' do
+      expect(addresses_api)
+        .to receive(:get_address_balance)
+        .with(wallet_id, address_id, 'eth')
+        .and_return(response)
       expect(address.get_balance(:eth)).to eq BigDecimal('1')
     end
 
     it 'returns the correct Gwei balance' do
+      expect(addresses_api)
+        .to receive(:get_address_balance)
+        .with(wallet_id, address_id, 'eth')
+        .and_return(response)
```

Comment:
Would probably simplify things and clean up the assertions to do this setup in a `before` block

```ruby
  before do
    allow(addresses_api)
        .to receive(:get_address_balance)
        .with(wallet_id, address_id, 'eth')
        .and_return(response)
  end
```